bindings/C/src/dbrPut.c

```diff
@@ -16,21 +16,23 @@
  */
 #include "libdbrAPI.h"
 
+#include <stdlib.h>
+
 DBR_Errorcode_t
 dbrPut (DBR_Handle_t cs_handle,
         void *va_ptr,
         int64_t size,
         DBR_Tuple_name_t tuple_name,
         DBR_Group_t group)
 {
-  dbBE_sge_t sge;
-  sge.iov_base = va_ptr;
-  sge.iov_len = size;
+  dbrDA_Request_chain_t *req = (dbrDA_Request_chain_t*)calloc( 1, sizeof( dbrDA_Request_chain_t ) + sizeof( dbBE_sge_t ) );;
```

Comment:
you really wanted this line to end..

Reply:
Yes, I wanted to make sure that the allocation terminates at some point. ;-)
It's also a reminder that this allocation is not optimal there. I have to fix that at a later point because it's happening for every request - I don't like that.